src/read.c

```diff
@@ -5711,7 +5711,9 @@ static avifResult avifDecoderDecodeTiles(avifDecoder * decoder, uint32_t nextIma
         }
 
         avifBool isLimitedRangeAlpha = AVIF_FALSE;
-        if (!tile->codec->getNextImage(tile->codec, decoder, sample, avifIsAlpha(tile->input->itemCategory), &isLimitedRangeAlpha, tile->image)) {
+        tile->codec->maxThreads = decoder->maxThreads;
+        tile->codec->imageSizeLimit = decoder->imageSizeLimit;
```

Comment:
To match the current behavior, I copy these two fields from `decoder` before each `getNextImage()` call.  In theory this allows users to change `decoder->maxThreads` or `decoder->imageSizeLimit` before decoding each image. Not sure if this actually works, because I believe all the codecs use these two fields in the first `getNextImage()` call only (when the underlying decoder instance is created).

An alternative is to copy these two fields only when `tile->codec` is created.